frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -371,84 +379,162 @@ function BookInfo:setCustomBookCover(file, book_props, metadata_updated_caller_c
-                        words_nb = words_nb + 1
-                    end
-                else
-                    words_nb = words_nb + 1
-                end
-            end
-        end
+function BookInfo:setCustomMetadata(file, book_props, metadata_updated_caller_callback, prop_key, prop_value)
+    -- in file
+    local custom_doc_settings, custom_props, display_title
+    if self.custom_doc_settings then
+        custom_doc_settings = self.custom_doc_settings
+        custom_props = custom_doc_settings:readSetting("custom_props")
+    else -- no custom metadata file, create new
+        custom_doc_settings = DocSettings:openCustomMetadata()
+        custom_props = {}
+        display_title = book_props.display_title -- backup
+        book_props.display_title = nil
+        custom_doc_settings:saveSetting("doc_props", book_props) -- save a copy of original props
+    end
+    custom_props[prop_key] = prop_value -- nil when resetting a custom prop
+    if next(custom_props) == nil then -- no more custom metadata
+        os.remove(custom_doc_settings.custom_metadata_file)
+        DocSettings:removeSidecarDir(file, util.splitFilePathName(custom_doc_settings.custom_metadata_file))
     else
-        local page_boxes = self.ui.document:getTextBoxes(self.ui:getCurrentPage())
-        if page_boxes and page_boxes[1][1].word then
-            lines_nb = #page_boxes
-            for _, line in ipairs(page_boxes) do
-                if #line == 1 and line[1].word == "" then -- empty line
-                    lines_nb = lines_nb - 1
-                else
-                    words_nb = words_nb + #line
-                    local last_word = line[#line].word
-                    if last_word:sub(-1) == "-" and last_word ~= "-" then -- hyphenated
-                        words_nb = words_nb - 1
-                    end
-                end
-            end
+        custom_doc_settings:saveSetting("custom_props", custom_props)
+        custom_doc_settings:flushCustomMetadata(file)
+    end
+    book_props.display_title = book_props.display_title or display_title -- restore
+    -- in memory
+    prop_value = prop_value or custom_doc_settings:readSetting("doc_props")[prop_key] -- set custom or restore original
+    book_props[prop_key] = prop_value
+    if self.ui then -- currently opened document
+        self.ui.doc_props[prop_key] = prop_value
+        if prop_key == "title" then -- generate if original is empty
+            self.ui.doc_props.display_title = prop_value or filemanagerutil.splitFileNameType(file)
         end
     end
-    return lines_nb, words_nb
+    self:updateBookInfo(file, book_props, metadata_updated_caller_callback, prop_key)
+end
+
+function BookInfo:showCustomEditDialog(file, book_props, metadata_updated_caller_callback, prop_key)
+    local input_dialog
+    input_dialog = InputDialog:new{
+        title = _("Edit book property: ") .. self.prop_text[prop_key]:sub(1, -2),
```

Comment:
`_("Title:"),` when translated in French (haven't checked) may be `_("Titre :")` with a space or thin-space before the `:`.
I guess no issue if there's a trailing space in the text of the title you make for that widget, but dunno about other languages and how they would translate these (and if we should include the `:` in the translatable strings or not (and have the code append it, but then French won't get that space).
@Frenzie : thoughts?

Reply:
Looks to me like it should be `Edit book property: %1` instead of trying to figure out how to concatenate things, but iff template isn't appropriate for some reason the space should be concatenated without bothering the translator with it, as in `Edit book property:` (colon at the end, no space at the end).

Reply:
My comment was more about https://github.com/koreader/koreader/pull/10861/files#diff-10106ca66f5be0b0a2554fc344220689d85fa930768116d5650e655c558f465bR32
the list of the keys of the Book info KVPage, which includes their `:`, and is reused there to make the title "Edit this property", without the there uneeded `:` .

Reply:
Ahh, there's a good chance that's being too clever yeah, although your French example is probably unproblematic (a tiny bit of extra space). But if Chinese uses some utf8 full width colon for example does it still work out...

Reply:
Fixed (replace ":" to "").